src/background.js

```diff
@@ -29,50 +29,89 @@ function sendVideoUrlToMetubeAndSwitchTab(videoUrl, metubeUrl, tab) {
         if (res.ok === true && res.status === 200) {
             return res.json();
         }
-        // todo fix it
-        alert("error :: code " + res.status);
+        console.log("error :: code" + res.status);
     }).then(function(result) {
-        if (result.status === "ok") {
-            openTab(metubeUrl, tab);
-        } else {
-            // todo fix it
-            alert("error :: " + json);
+        if (result.status !== "ok") {
+            console.log("error :: ", result );
+            if (result.msg.includes('URLError')) {
+                // Go straight to catch block and skip the callback
+                throw new Error('Error when adding URL to MyTube');
+            }
+        }
+    }).then(function() {
+        if (typeof callback === 'function'){
+            callback();
         }
-    }).catch(function(res) {
-        alert("error :: " + res);
+    }).catch(function(e) {
+        console.log("Ran into an error :: ", e);
     });
 }
 
 chrome.contextMenus.onClicked.addListener(function(item, tab) {
-    chrome.storage.sync.get(['metube'], function(data) {
+    chrome.storage.sync.get(['metube', 'contextMenuClickBehavior'], function(data) {
         if (data === undefined || !data.hasOwnProperty('metube') || data.metube === "") {
             openTab(chrome.runtime.getURL('options.html'), tab);
             return
         }
-        sendVideoUrlToMetubeAndSwitchTab(item.linkUrl, data.metube, tab);
+        if ( data.contextMenuClickBehavior == 'context-menu-send-current-url') {
+            sendVideoUrlToMetube(item.linkUrl, data.metube);
+        } else if (data.contextMenuClickBehavior == 'context-menu-send-current-url-and-switch'){
+            sendVideoUrlToMetube(item.linkUrl, data.metube, function() {
+                openTab(data.metube, tab);
+            });
+        } else {
+            console.log("Unknown contextMenuClickBehavior value: " + data.contextMenuClickBehavior);
+        }
     });
 });
 
+
+async function getCurrentTabUrl() {
```

Comment:
i can't find usage of this function. we are realy need it?

Reply:
Yeah, I incorrectly committed this - removing